recipes/ignition-cmake/all/conanfile.py

```diff
@@ -0,0 +1,57 @@
+import glob
+import os
+
+from conans import CMake, ConanFile, tools
+
+
+class IgnitionCmakeConan(ConanFile):
+    name = "ignition-cmake"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ignitionrobotics/ign-cmake"
+    description = "A set of CMake modules that are used by the C++-based Ignition projects."
+    topics = ("ignition", "robotics", "cmake")
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def package_id(self):
+        self.info.header_only()
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_INSTALL_DATAROOTDIR"] = "lib"
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(glob.glob("ign-cmake*")[0], self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        version_major = tools.Version(self.version).major
+        self.cpp_info.names["cmake_find_package"] = "ignition-cmake{}".format(version_major)
+        self.cpp_info.names["cmake_find_package_multi"] = "ignition-cmake{}".format(version_major)
+        self.cpp_info.builddirs = [
+            os.path.join("lib", "cmake", "ignition-cmake{}".format(version_major)),
+            os.path.join("lib", "cmake", "ignition-cmake{}".format(version_major), "cmake{}".format(version_major)),
+        ]
```

Comment:
You have to pass build_modules:

https://github.com/conan-io/conan-center-index/blob/6fbbfe576054a57cc845f008ec79ef2934fc3f79/recipes/pybind11/all/conanfile.py#L53

Reply:
I tried with:
```python
    def package_info(self):
        version_major = tools.Version(self.version).major
        self.cpp_info.names["cmake_find_package"] = "ignition-cmake{}".format(version_major)
        self.cpp_info.names["cmake_find_package_multi"] = "ignition-cmake{}".format(version_major)

        cmake_base_path = os.path.join("lib", "cmake", "ignition-cmake{}".format(version_major), "cmake{}".format(version_major))

        self.cpp_info.builddirs = [
            os.path.join("lib", "cmake", "ignition-cmake{}".format(version_major)),
            cmake_base_path,
        ]

        for cmake_file in glob.glob("{}/*.cmake".format(cmake_base_path)):
            self.cpp_info.build_modules.append(cmake_file)
```
But adding the last `for` loop results in an error during the `test_package`:
```
CMake Error at /home/juan/.conan/data/ignition-cmake/2.5.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/ignition-cmake2/cmake2/IgnPkgConfig.cmake:75 (_ign_cmake_parse_arguments):
  Unknown CMake command "_ign_cmake_parse_arguments".
Call Stack (most recent call first):
  /home/juan/.conan/data/ignition-cmake/2.5.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/ignition-cmake2/cmake2/FindAVCODEC.cmake:19 (ign_pkg_check_modules_quiet)
  build/8e7da137254ca61c29cfd61de2482b72ba0ff99c/conanbuildinfo.cmake:857 (include)
  build/8e7da137254ca61c29cfd61de2482b72ba0ff99c/conanbuildinfo.cmake:335 (conan_include_build_modules)
  CMakeLists.txt:5 (conan_basic_setup)


-- Configuring incomplete, errors occurred!
```